src/javascript-extract.js

```diff
@@ -104,8 +127,8 @@ function getGettextEntriesFromScript(script) {
   return extractedEntries;
 }
 
-function extractStringsFromJavascript(filename, script) {
-  const gettextEntries =  getGettextEntriesFromScript(script);
+function extractStringsFromJavascript(filename, script, parser = 'auto') {
+  const gettextEntries =  getGettextEntriesFromScript(script, parser);
```

Comment:
I think you could avoid adding a `parser` argument to `getGettextEntriesFromScript` and just directly call one or both of the parsing submethods (let's call them `parseJSGettextWithAcorn()` and `parseJSGettextWithBabel()`  directly here. It would make a clearer separation of abstraction levels.